src/phpcs/Production/Sniffs/Strings/DisallowDoubleQuoteUsageSniff.php

```diff
@@ -50,42 +59,54 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPointer)
         // The use of variables in double quoted strings is not allowed.
         if ($tokens[$stackPointer]['code'] === T_DOUBLE_QUOTED_STRING)
         {
-            $stringTokens = token_get_all('<?php ' . $workingString);
-            foreach ($stringTokens as $token)
+            $this->handleDoubleQuotedString($phpcsFile, $stackPointer, $workingString);
+        }
+
+        $this->handleSpecialChars($phpcsFile, $stackPointer, $workingString);
+    }
+
+    /**
+     * This method adds errors if double quoted strings contain variables.
+     *
+     * @param File   $phpcsFile
+     * @param int    $stackPointer
+     * @param string $workingString
+     *
+     * @return void
+     */
+    protected function handleDoubleQuotedString(File $phpcsFile, $stackPointer, $workingString)
+    {
+        $stringTokens = token_get_all('<?php ' . $workingString);
+        foreach ($stringTokens as $token)
+        {
+            if (is_array($token) && $token[0] === T_VARIABLE)
             {
-                if (is_array($token) && $token[0] === T_VARIABLE)
-                {
-                    $error = 'Variable "%s" not allowed in double quoted string; use concatenation and single quotes instead';
-                    $data  = [$token[1]];
-                    $phpcsFile->addError($error, $stackPointer, 'ContainsVar', $data);
-                }
+                $error = 'Variable "%s" not allowed in double quoted string; use concatenation and single quotes instead';
+                $data  = [$token[1]];
+                $phpcsFile->addError($error, $stackPointer, 'Prduction.DisallowDoubleQuoteUsage.ContainsVar', $data);
             }
-
-            return;
         }
+    }
 
-        $specialChars = [
-            '\n',
-            '\r',
-            '\f',
-            '\t',
-        ];
-
-        $errorThrown = false;
-        foreach ($specialChars as $testChar)
+    /**
+     * This method adds errors if double quoted strings contain control characters like \n.
+     *
+     * @param File   $phpcsFile
+     * @param int    $stackPointer
+     * @param string $workingString
+     *
+     * @return void
+     */
+    protected function handleSpecialChars(File $phpcsFile, $stackPointer, $workingString)
+    {
+        foreach ($this->specialChars as $testChar)
         {
             if (strpos($workingString, $testChar) !== false)
             {
                 $error    = 'Please use constants (LF, CR, CRLF, ...) using concatenation instead of "%s" in double quotes';
```

Comment:
How is the new line character expressed, when used within a sprintf parameter?

Reply:
You will see "\n" ^^